### PR TITLE
add a check for env variable RELEASE_CHANNEL while generating docs

### LINKED ISSUE #77178: api.flutter.dev docs footer says "HEAD (no branch)"
This manifests in the following issues:

- api.flutter.dev docs footer says "HEAD (no branch)"
- dartpad generated samples on stable to load the wrong code: https://github.com/flutter/flutter/issues/86297

This is a regression from when we migrated from Cirrus to LUCI. This is affecting both [stable docs](https://api.flutter.dev) and [master docs](https://master-api.flutter.dev/).

We are running `git status -b --porcelain` ([link](https://github.com/flutter/flutter/blob/master/dev/tools/dartdoc.dart#L257)) to determine the branch name, but it seems that on LUCI we are checking out a specific commit, resulting in a detached HEAD (i.e. no branch). I think the easiest way to fix this would be to update the dartdoc script to read an env variable, and ensure the recipe provides it.

### PR DESCRIPTION
**Pull request by christopherfujino**
Fixes: https://github.com/flutter/flutter/issues/77178

The env variable `LUCI_BRANCH` is already provided during this build, as seen in: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8841621283261717792/+/u/prepare_environment/flutter_doctor/execution_details

It gets set here: https://flutter.googlesource.com/recipes/+/refs/heads/master/recipe_modules/repo_util/api.py#144

### CODE REVIEW

**Comment by gspencergoog** [comment]
Looks good, but can you add a test for this?

**Comment by christopherfujino** [comment]
> Looks good, but can you add a test for this?

Done, PTAL

File: dev/snippets/pubspec.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ dependencies:
   dart_style: 2.0.2
   meta: 1.7.0
   platform: 3.0.0
+  process: 4.2.1
```
Comment (gspencergoog) [inline dev/snippets/pubspec.yaml:18]: Don't you also need to update all of the other pubspecs with `flutter update-packages --force-upgrade`? Or did you do that, and there just weren't any updates?

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: dev/snippets/pubspec.yaml (hunk 1 above)
Comment (christopherfujino) [reply dev/snippets/pubspec.yaml:18]: i did

**Pull request merged**